lib/identity_cache/configuration_dsl.rb

```diff
@@ -291,9 +291,13 @@ def ensure_cacheable_association(association, options)
         if association_reflection.options[:through]
           raise UnsupportedAssociationError, "caching through associations isn't supported"
         end
+        child_class = association_reflection.klass
+        scope = association_reflection.scope
+        if scope && !child_class.instance_exec(&scope).joins_values.empty?
```

Comment:
I prefer to use `.empty?` rather than `.blank?` or `.present?` unless I want to handle the receiver being `nil`.